File: functions/entity/payment.js
// payments 컬렉션 (결제 내역)
// 문서 ID: paymentId (자동 생성)
// {
//   payer_id: string, // 결제한 사용자 (여행객)
//   receiver_id: string, // 결제를 받는 현지인 (해당 없으면 null)
//   type: 'guide_fee' | 'reservation_fee',
//   amount: number,
//   status: 'completed' | 'pending' | 'failed' | 'refunded',
//   method: 'card' | 'kakaopay' | 'naverpay',
//   related_matching_id?: string,
//   related_guide_id?: string,
//   paid_at: string, // ISO8601
//   created_at: string, // ISO8601
//   updated_at: string // ISO8601
// }

class Payment {
  constructor(data) {
    this.payer_id = data.payer_id;
    this.receiver_id = data.receiver_id;
    this.type = data.type;
    this.amount = data.amount;
    this.status = data.status;
    this.method = data.method;
    this.related_matching_id = data.related_matching_id;
    this.related_guide_id = data.related_guide_id;
    this.paid_at = data.paid_at;
    this.created_at = data.created_at;
    this.updated_at = data.updated_at;
  }
}
module.exports = Payment; 

File: app/lib/models/commerce/payment.dart
class Payment {
  final String payerId;
  final String receiverId;
  final String type;
  final int amount;
  final String status;
  final String method;
  final String? relatedMatchingId;
  final String? relatedGuideId;
  final DateTime paidAt;
  final DateTime createdAt;
  final DateTime updatedAt;

  Payment({
    required this.payerId,
    required this.receiverId,
    required this.type,
    required this.amount,
    required this.status,
    required this.method,
    this.relatedMatchingId,
    this.relatedGuideId,
    required this.paidAt,
    required this.createdAt,
    required this.updatedAt,
  });

  factory Payment.fromJson(Map<String, dynamic> json) => Payment(
        payerId: json['payer_id'],
        receiverId: json['receiver_id'],
        type: json['type'],
        amount: json['amount'],
        status: json['status'],
        method: json['method'],
        relatedMatchingId: json['related_matching_id'],
        relatedGuideId: json['related_guide_id'],
        paidAt: DateTime.parse(json['paid_at']),
        createdAt: DateTime.parse(json['created_at']),
        updatedAt: DateTime.parse(json['updated_at']),
      );

  Map<String, dynamic> toJson() => {
        'payer_id': payerId,
        'receiver_id': receiverId,
        'type': type,
        'amount': amount,
        'status': status,
        'method': method,
        'related_matching_id': relatedMatchingId,
        'related_guide_id': relatedGuideId,
        'paid_at': paidAt.toIso8601String(),
        'created_at': createdAt.toIso8601String(),
        'updated_at': updatedAt.toIso8601String(),
      };
}
